src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGANaivelySpark.java

```diff
@@ -0,0 +1,485 @@
+     *                                   and the second is another tuple of the temporary directory and a temporary FASTQ file
+     *                                   containing sequence information.
+     * @throws IOException               an IOException is thrown if the conversion fails to create the FASTQ file
+     */
+    @VisibleForTesting
+    static Tuple2<Long, File> convertReadsToFASTQ(final Tuple2<Long, Iterable<GATKRead>> orderedReadsOfABreakPoint) throws IOException{
+
+        final Long breakpointId = orderedReadsOfABreakPoint._1();
+
+        File workingDir = new File( Files.createTempDirectory("assembly" + orderedReadsOfABreakPoint._1().toString()).toString() );
+        if(null==workingDir){
+            throw new IOException("Failed to create temporary directory to perform assembly in.");
+        }
+        IOUtils.deleteRecursivelyOnExit(workingDir);
+
+        final File fastq = new File(workingDir, "assembly" + breakpointId + ".fastq");
+        if(null==fastq){
+            throw new IOException("Failed to create temporary FASTQ file to feed to SGA.");
+        }
+
+        try(final FastqWriter writer = new FastqWriterFactory().newWriter(fastq)){
+            Iterator<GATKRead> it = orderedReadsOfABreakPoint._2().iterator();
+            while(it.hasNext()){
+                writeToTempFASTQ(it.next(), writer);
+                writeToTempFASTQ(it.next(), writer);
+            }
+        }
+
+        return new Tuple2<>(breakpointId, fastq);
+    }
+
+    // copied from FindSVBreakpointsSpark
+    private static void writeToTempFASTQ(final GATKRead read, final FastqWriter writer){
+        final String readName      = read.getName();// TODO: hack for unit testing, where read names have /1 and /2. should be:   + (read.isSecondOfPair() ? "/2" : "/1");
+        final String readBases     = read.getBasesString();
+        final String baseQualities = ReadUtils.getBaseQualityString(read);
+        writer.write(new FastqRecord(readName, readBases, "", baseQualities));
+    }
+
+    /**
+     * Final return type of the whole process of SGA local assembly.
+     * assembledContigFile is the file containing the assembled contigs, if the process executed successfully, or null if not.
+     * runtimeInformation contains the runtime information logged along the process up until the process erred, if errors happen,
+     *  or until the last step if no errors occur along the line.
+     *  The list is organized along the process of executing the assembly pipeline, inside each entry of the list is
+     *      the name of the module,
+     *      return status of the module,
+     *      stdout message of the module, if any,
+     *      stderr message of the module, if any
+     */
+    @VisibleForTesting
+    static final class SGAAssemblyResultType implements Serializable{
+        private static final long serialVersionUID = 1L;
+
+        public final File assembledContigFile;
```

Comment:
Rather than just keeping a link to the file, shouldn't this class actually read the contents of the file into memory, so they can be in the RDD? I guess it depends somewhat on what we want to do next with the data -- if we are just going to write it back out to file to run BWA on it, it doesn't make much sense to read it back in. But in that case we probably shouldn't map it back into an RDD, because again we have paths to local files being passed around in a distributed context.